In [1]:
import pandas as pd
import numpy as np

In [2]:
pd.read_csv('./data/ThoraricSurgery.csv')

293  1   3.8   2.8  0  0.1  0.2  0.3  0.4  0.5  12  0.6  0.7  0.8  1.1  \
0      1  2  2.88  2.16  1    0    0    0    1    1  14    0    0    0    1   
1      8  2  3.19  2.50  1    0    0    0    1    0  11    0    0    1    1   
2     14  2  3.98  3.06  2    0    0    0    1    1  14    0    0    0    1   
3     17  2  2.21  1.88  0    0    1    0    0    0  12    0    0    0    1   
4     18  2  2.96  1.67  0    0    0    0    0    0  12    0    0    0    1   
..   ... ..   ...   ... ..  ...  ...  ...  ...  ...  ..  ...  ...  ...  ...   
464   98  6  3.04  2.40  2    0    0    0    1    0  11    0    0    0    1   
465  369  6  3.88  2.72  1    0    0    0    1    0  12    0    0    0    1   
466  406  6  5.36  3.96  1    0    0    0    1    0  12    0    0    0    0   
467   25  8  4.32  3.20  0    0    0    0    0    0  11    0    0    0    0   
468  447  8  5.20  4.10  0    0    0    0    0    0  12    0    0    0    0   

     0.9  62  0.10  
0      0  60     0  
1      0  66     1  
2      0  80     1  
3      0  56     0  
4      0  61     0  
..   ...  ..   ...  
464    0  76     0  
465    0  77     0  
466    0  62     0  
467    0  58     1  
468    0  49     0  

[469 rows x 18 columns]

In [3]:
data = np.loadtxt('./data/ThoraricSurgery.csv', delimiter=',')

In [4]:
data

array([[293.  ,   1.  ,   3.8 , ...,   0.  ,  62.  ,   0.  ],
       [  1.  ,   2.  ,   2.88, ...,   0.  ,  60.  ,   0.  ],
       [  8.  ,   2.  ,   3.19, ...,   0.  ,  66.  ,   1.  ],
       ...,
       [406.  ,   6.  ,   5.36, ...,   0.  ,  62.  ,   0.  ],
       [ 25.  ,   8.  ,   4.32, ...,   0.  ,  58.  ,   1.  ],
       [447.  ,   8.  ,   5.2 , ...,   0.  ,  49.  ,   0.  ]])

In [5]:
data = pd.DataFrame(data)

In [6]:
data

0    1     2     3    4    5    6    7    8    9     10   11   12  \
0    293.0  1.0  3.80  2.80  0.0  0.0  0.0  0.0  0.0  0.0  12.0  0.0  0.0   
1      1.0  2.0  2.88  2.16  1.0  0.0  0.0  0.0  1.0  1.0  14.0  0.0  0.0   
2      8.0  2.0  3.19  2.50  1.0  0.0  0.0  0.0  1.0  0.0  11.0  0.0  0.0   
3     14.0  2.0  3.98  3.06  2.0  0.0  0.0  0.0  1.0  1.0  14.0  0.0  0.0   
4     17.0  2.0  2.21  1.88  0.0  0.0  1.0  0.0  0.0  0.0  12.0  0.0  0.0   
..     ...  ...   ...   ...  ...  ...  ...  ...  ...  ...   ...  ...  ...   
465   98.0  6.0  3.04  2.40  2.0  0.0  0.0  0.0  1.0  0.0  11.0  0.0  0.0   
466  369.0  6.0  3.88  2.72  1.0  0.0  0.0  0.0  1.0  0.0  12.0  0.0  0.0   
467  406.0  6.0  5.36  3.96  1.0  0.0  0.0  0.0  1.0  0.0  12.0  0.0  0.0   
468   25.0  8.0  4.32  3.20  0.0  0.0  0.0  0.0  0.0  0.0  11.0  0.0  0.0   
469  447.0  8.0  5.20  4.10  0.0  0.0  0.0  0.0  0.0  0.0  12.0  0.0  0.0   

      13   14   15    16   17  
0    0.0  1.0  0.0  62.0  0.0  
1    0.0  1.0  0.0  60.0  0.0  
2    1.0  1.0  0.0  66.0  1.0  
3    0.0  1.0  0.0  80.0  1.0  
4    0.0  1.0  0.0  56.0  0.0  
..   ...  ...  ...   ...  ...  
465  0.0  1.0  0.0  76.0  0.0  
466  0.0  1.0  0.0  77.0  0.0  
467  0.0  0.0  0.0  62.0  0.0  
468  0.0  0.0  0.0  58.0  1.0  
469  0.0  0.0  0.0  49.0  0.0  

[470 rows x 18 columns]

- 폴란드 브로츠와프 의과대학에서 2013년 공개한 폐암 수술환자의 수술 전 데이터와 수술 후 생존결과를 기록한 의료기록 데이터
- 18개 컬럼으로 구성, 470개의 데이터, 각 항목은 쉼표(,)로 구분
- 종양 유형, 폐활량, 호흡곤란여부, 고통 정도, 기침, 흡연, 천식여부 등 17가지 환자 상태
- 18번째 컬럼은 수술 후 생존 결과 (<span style="color:green;">1 : 생존, 0 : 사망</span>)


In [7]:
X = data.iloc[:,:-1]
y = data.iloc[:,-1]
X

0    1     2     3    4    5    6    7    8    9     10   11   12  \
0    293.0  1.0  3.80  2.80  0.0  0.0  0.0  0.0  0.0  0.0  12.0  0.0  0.0   
1      1.0  2.0  2.88  2.16  1.0  0.0  0.0  0.0  1.0  1.0  14.0  0.0  0.0   
2      8.0  2.0  3.19  2.50  1.0  0.0  0.0  0.0  1.0  0.0  11.0  0.0  0.0   
3     14.0  2.0  3.98  3.06  2.0  0.0  0.0  0.0  1.0  1.0  14.0  0.0  0.0   
4     17.0  2.0  2.21  1.88  0.0  0.0  1.0  0.0  0.0  0.0  12.0  0.0  0.0   
..     ...  ...   ...   ...  ...  ...  ...  ...  ...  ...   ...  ...  ...   
465   98.0  6.0  3.04  2.40  2.0  0.0  0.0  0.0  1.0  0.0  11.0  0.0  0.0   
466  369.0  6.0  3.88  2.72  1.0  0.0  0.0  0.0  1.0  0.0  12.0  0.0  0.0   
467  406.0  6.0  5.36  3.96  1.0  0.0  0.0  0.0  1.0  0.0  12.0  0.0  0.0   
468   25.0  8.0  4.32  3.20  0.0  0.0  0.0  0.0  0.0  0.0  11.0  0.0  0.0   
469  447.0  8.0  5.20  4.10  0.0  0.0  0.0  0.0  0.0  0.0  12.0  0.0  0.0   

      13   14   15    16  
0    0.0  1.0  0.0  62.0  
1    0.0  1.0  0.0  60.0  
2    1.0  1.0  0.0  66.0  
3    0.0  1.0  0.0  80.0  
4    0.0  1.0  0.0  56.0  
..   ...  ...  ...   ...  
465  0.0  1.0  0.0  76.0  
466  0.0  1.0  0.0  77.0  
467  0.0  0.0  0.0  62.0  
468  0.0  0.0  0.0  58.0  
469  0.0  0.0  0.0  49.0  

[470 rows x 17 columns]

In [8]:
y

0      0.0
1      0.0
2      1.0
3      1.0
4      0.0
      ... 
465    0.0
466    0.0
467    0.0
468    1.0
469    0.0
Name: 17, Length: 470, dtype: float64

In [9]:
from sklearn.model_selection import train_test_split

In [10]:
X_train, X_test, y_train, y_test = train_test_split(X, y, 
                                                    test_size = 0.3,
                                                    random_state = 1)

In [11]:
print(X_train.shape)
print(X_test.shape)
print(y_train.shape)
print(y_test.shape)

(329, 17)
(141, 17)
(329,)
(141,)


In [12]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier

In [13]:
knn = KNeighborsClassifier()
tree = DecisionTreeClassifier()

In [14]:
knn.fit(X_train, y_train)
tree.fit(X_train, y_train)

DecisionTreeClassifier()

In [15]:
knn.score(X_test, y_test)

0.8439716312056738

In [16]:
tree.score(X_test, y_test)

0.7446808510638298

In [17]:
X_test.iloc[0, :]

0     208.00
1       3.00
2       2.60
3       1.92
4       1.00
5       0.00
6       0.00
7       0.00
8       1.00
9       0.00
10     11.00
11      0.00
12      0.00
13      0.00
14      1.00
15      0.00
16     66.00
Name: 207, dtype: float64

In [25]:
knn.predict([X_test.iloc[0,:]])

array([0.])